### ISSUE TITLE
kotlinx - UUID Serializer - String

### ISSUE THREAD

**Issue by NitrinCloud** [Litote/kmongo#411]
When I'm trying to encode a data class to a string with an UUID inside I get...

`Serializer for class 'UUID' is not found.`

Is there a way to use the internal serializer for uuid to get the object as a string and decode the string back to it? Or any other way I can still use the UuidRepresentation

**Comment by zigzago** [comment]
Does this unit test fill your needs ? https://github.com/Litote/kmongo/blob/master/kmongo-core-tests/src/main/kotlin/org/litote/kmongo/issues/Issue245UuidRepresentation.kt#L44

**Comment by NitrinCloud** [comment]
Not really. Currently I use MongoDB as my main database to store user data. I need to retrieve the data very often and on multiple servers, so I thought I just use a cache like redis since it is much faster. 

In redis you can just insert strings or byte data, so I need to be able to encode and decode the data class to/from a string.

I found [this](https://github.com/Litote/kmongo/blob/master/kmongo-shared/src/main/kotlin/org/litote/kmongo/SharedExtensions.kt) but I found no way to decode the string back to the data class object

**Comment by zigzago** [comment]
This is more a serialization issue then. 

You can use CodecRegistry:
```kotlin

val registry = KMongoClient.createClient().getDatabase("test").codecRegistry
val codec = registry.getCodec(MyClass::class.java)
val myClass  = codec.decode(BsonDocument.parse(json), DecoderContext.builder().build())
```

HTH
